goal_src/levels/jungleb/plat-flip.gc

```diff
@@ -5,3 +5,419 @@
+          s5-1
+          (the-as art-joint-anim (-> self draw art-group data 2))
+          num-func-identity
+          )
+         (set!
+          (-> s5-1 frame-num)
+          (/
+           (*
+            f30-0
+            (the
+             float
+             (+
+              (->
+               (the-as art-joint-anim (-> self draw art-group data 2))
+               data
+               0
+               length
+               )
+              -1
+              )
+             )
+            )
+           (-> self turn-down-time)
+           )
+          )
+         )
+        )
+       ((begin
+         (set! f30-1 (- f30-0 (-> self turn-down-time)))
+         (< f30-1 (-> self before-turn-up-time))
+         )
+        (set! gp-0 #f)
+        (let ((s5-2 (-> self skel root-channel 0)))
+         (joint-control-channel-group-eval!
+          s5-2
+          (the-as art-joint-anim (-> self draw art-group data 2))
+          num-func-identity
+          )
+         (set!
+          (-> s5-2 frame-num)
+          (the
+           float
+           (+
+            (->
+             (the-as art-joint-anim (-> self draw art-group data 2))
+             data
+             0
+             length
+             )
+            -1
+            )
+           )
+          )
+         )
+        )
+       (else
+        (when (not gp-0)
+         (set! gp-0 #t)
+         (sound-play-by-name
+          (static-sound-name "plat-flip")
+          (new-sound-id)
+          1024
+          0
+          0
+          (the-as uint 1)
+          (the-as vector #t)
+          )
+         )
+        (let ((f30-2 (- f30-1 (-> self before-turn-up-time)))
+              (s5-4 (-> self skel root-channel 0))
+              )
+         (joint-control-channel-group-eval!
+          s5-4
+          (the-as art-joint-anim (-> self draw art-group data 3))
+          num-func-identity
+          )
+         (set!
+          (-> s5-4 frame-num)
+          (/
+           (*
+            f30-2
+            (the
+             float
+             (+
+              (->
+               (the-as art-joint-anim (-> self draw art-group data 3))
+               data
+               0
+               length
+               )
+              -1
+              )
+             )
+            )
+           (-> self turn-up-time)
+           )
+          )
+         )
+        )
+       )
+      )
+     (let ((s5-5 (new 'stack-no-clear 'vector)))
+      (set! (-> s5-5 quad) (-> self base-pos quad))
+      (+! (-> s5-5 y) (* 1638.4 (update! (-> self smush))))
+      (TODO-RENAME-30 (-> self root-override) s5-5)
+      )
+     (suspend)
+     )
+    )
+   (none)
+   )
+  :post
+  (the-as (function none :behavior plat-flip) rider-post)
+  )
+
+;; definition for method 11 of type plat-flip
+;; Used lq/sq
+(defmethod copy-defaults! plat-flip ((obj plat-flip) (arg0 res-lump))
+  (local-vars
+   (r0-0 uint128)
+   (a0-17 uint128)
+   (a0-26 uint128)
+   (a1-11 uint128)
+   (sv-16 int)
```

Comment:
sv-16 needs to be a res-tag here, otherwise this stack variable will be too small.  You can add a `type_casts.json` entry:
```json
["_stack_", 16, "res-tag"],
```

Reply:
oh interesting, ill have to keep my eye out for this